test/Tests/Writers/Typst.hs

```diff
@@ -0,0 +1,178 @@
+                    [TableBody nullAttr 0 [] $ map toRow rows]
+                    (TableFoot nullAttr [])
+    =?> unlines [
+        "#figure("
+      , "  align(center)[#text(fill: orange)[#table("
+      , "    columns: 0,"
+      , "    align: (),"
+      , "  )]]"
+      , "  , kind: table"
+      , "  )"
+      ]
+    , "table cell" =:
+      let headers = []
+          rows = [Row nullAttr [
+              cellWith ("", [], [("typst:fill", "blue")]) AlignDefault (RowSpan 1) (ColSpan 1) $ para "Foo"
+            , simpleCell $ para "Bar"
+            ]]
+      in table
+                    emptyCaption
+                    [(AlignDefault,ColWidthDefault),(AlignDefault,ColWidthDefault)]
+                    (TableHead nullAttr $ toHeaderRow headers)
+                    [TableBody nullAttr 0 [] rows]
+                    (TableFoot nullAttr [])
+    =?> unlines [
+        "#figure("
+      , "  align(center)[#table("
+      , "    columns: 2,"
+      , "    align: (auto,auto,),"
+      , "    table.cell(fill: blue)[Foo"
+      , ""
+      , "    ], [Bar"
+      , ""
+      , "    ],"
+      , "  )]"
+      , "  , kind: table"
+      , "  )"
+      ]
+    , "table cell text no attr" =:
+      let headers = []
+          rows = [Row nullAttr [
+              cellWith ("", [], [("typst:text:fill", "orange")]) AlignDefault (RowSpan 1) (ColSpan 1) $ para "Foo"
+            , simpleCell $ para "Bar"
+            ]]
+      in table
+                    emptyCaption
+                    [(AlignDefault,ColWidthDefault),(AlignDefault,ColWidthDefault)]
+                    (TableHead nullAttr $ toHeaderRow headers)
+                    [TableBody nullAttr 0 [] rows]
+                    (TableFoot nullAttr [])
+    =?> unlines [
+        "#figure("
+      , "  align(center)[#table("
+      , "    columns: 2,"
+      , "    align: (auto,auto,),"
+      , "    [#set text(fill: orange); Foo"
+      , ""
+      , "    ], [Bar"
+      , ""
+      , "    ],"
+      , "  )]"
+      , "  , kind: table"
+      , "  )"
+      ]
+    , "table cell text w attr" =:
+      let headers = []
+          rows = [Row nullAttr [
+              cellWith ("", [], [("typst:text:fill", "orange")]) AlignCenter (RowSpan 1) (ColSpan 1) $ para "Foo"
+            , simpleCell $ para "Bar"
+            ]]
+      in table
+                    emptyCaption
+                    [(AlignDefault,ColWidthDefault),(AlignDefault,ColWidthDefault)]
+                    (TableHead nullAttr $ toHeaderRow headers)
+                    [TableBody nullAttr 0 [] rows]
+                    (TableFoot nullAttr [])
+    =?> unlines [
+        "#figure("
+      , "  align(center)[#table("
+      , "    columns: 2,"
+      , "    align: (auto,auto,),"
+      , "    table.cell(align: center)[#set text(fill: orange); Foo"
+      , ""
+      , "    ], [Bar"
+      , ""
+      , "    ],"
+      , "  )]"
+      , "  , kind: table"
+      , "  )"
+      ]
+    ]
+  ]
```

Comment:
I'd suggest just using a "command test" (see `test/commands/*.md`).
They are easier to create and much easier to maintain (since we can just do `make test TESTARGS=--accept` to update the expected output instead of doing it manually.

Reply:
Aha, much simpler to write tests that way!

I threw in negative cases: elements that don't have `typst:*` attributes should not affect the output. This was covered by previous tests but worth it to be somewhat complete.